src/filesystem/impls/filer/ArchiveUtils.js

```diff
@@ -194,15 +197,32 @@ define(function (require, exports, module) {
             });
         }
 
-        add(root, function(err) {
-            if(err) {
-                return callback(err);
+        // Check if path is a directory or not
+        fs.stat(path, function(err, stats) {
+            if (stats.type === "DIRECTORY") {
+                add(path, function(err) {
+                    if(err) {
+                        return callback(err);
+                    }
+                    // Prepare folder for download
+                    var compressed = jszip.generate({type: 'arraybuffer'});
+                    var blob = new Blob([compressed], {type: "application/zip"});
+                    saveAs(blob, "project.zip");
+                    callback();
+                });
+            } else {
+                // Prepare file for download
+                fs.readFile(path, {encoding: null}, function(err, data){
+                    if (err) {
+                       return callback(err);
+                    }
+                    
+                    var blob = new Blob([data], {type: "text/plain"});
```

Comment:
use https://github.com/mozilla/brackets/blob/master/src/filesystem/impls/filer/lib/content.js#L40 for the mime-type.